### PR TITLE
[chore] Migrate the "generate" make target to mdatagen from core repo

### PR DESCRIPTION
**Pull request by dmitryax**
This change migrates `generate` make target from using the deprecated `cmd/mdatagen` in this repository to mdategen defined in core repository.

To avoid breaking changes for the end users, we keep the scope names used in this repo as before. This required defining them explicitly in metadata.yaml files. We can update them after https://github.com/open-telemetry/opentelemetry-collector/issues/9494 and https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469 are resolved.

Taking the opportunity that the scope names can be explicitly defined, this PR also updates missing scope names for extensions with inconsistent package names e.g.: `awsproxy` and `jaegerremotesampling`. It's not a breaking change because the generated meter and tracer are not being used yet.

This change unblocks https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30495

### CODE REVIEW

**Comment by dmitryax** [comment]
failing tests should be fixed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31610

File: internal/tools/go.mod [hunk 1]
```diff
@@ -16,6 +16,7 @@ require (
 	go.opentelemetry.io/build-tools/issuegenerator v0.12.0
 	go.opentelemetry.io/build-tools/multimod v0.13.0
 	go.opentelemetry.io/collector/cmd/builder v0.96.0
+	go.opentelemetry.io/collector/cmd/mdatagen v0.96.1-0.20240304230612-89dcb871686b
```
Comment (MovieStoreGuy) [inline internal/tools/go.mod:19]: Do we want this to be the same version used by the builder?

Comment (dmitryax) [reply internal/tools/go.mod:19]: Yeah, I'm actually doing the core upgrade in a separate PR

Comment (dmitryax) [reply internal/tools/go.mod:19]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31612

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**